src/cbor.c

```diff
@@ -1132,6 +1132,64 @@ decode_attcred(const unsigned char **buf, size_t *len, int cose_alg,
 	return (ok);
 }
 
+static int
+decode_attobj(const cbor_item_t *key, const cbor_item_t *val, void *arg)
+{
+	fido_cred_t *cred = arg;
+	char *name = NULL;
+	int ok = -1;
+
+	if (cbor_string_copy(key, &name) < 0) {
+		fido_log_debug("%s: cbor type", __func__);
+		ok = 0; /* ignore */
+		goto fail;
+	}
+
+	if (!strcmp(name, "fmt")) {
+		if (cbor_decode_fmt(val, &cred->fmt) < 0) {
+			fido_log_debug("%s: cbor_decode_fmt", __func__);
+			goto fail;
+		}
+	} else if (!strcmp(name, "attStmt")) {
+		if (cbor_decode_attstmt(val, &cred->attstmt) < 0) {
```

Comment:
something we need to reevaluate at one point is how we deal with duplicate entries in maps, which IIRC are allowed in CTAP CBOR. in some places (like cbor_decode_fmt) we error out, in others (cbor_decode_attstmt, alg case) we silently overwrite previously observed values

Reply:
Mainly as a reference to myself:

> something we need to reevaluate at one point is how we deal with duplicate entries in maps, which IIRC are allowed in CTAP CBOR

[§ 8. Message Encoding](https://fidoalliance.org/specs/fido-v2.1-ps-20210615/fido-client-to-authenticator-protocol-v2.1-ps-20210615.html#message-encoding) mentions

> All decoders SHOULD reject CBOR that is not validly encoded in the [CTAP2 canonical CBOR encoding form](https://fidoalliance.org/specs/fido-v2.1-ps-20210615/fido-client-to-authenticator-protocol-v2.1-ps-20210615.html#ctap2-canonical-cbor-encoding-form) and SHOULD reject messages with duplicate map keys.

> in others (cbor_decode_attstmt, alg case) we silently overwrite previously observed values

Yes, `alg` is the odd one out there. We could certainly add a demanding `COSE_UNSPEC`.